### PR TITLE
Market decentralized/cleanup

### PR DESCRIPTION
**Pull request by tworec**
Cleanup arose when coding #333:
- [x] subscription id disambiguation
- [x] testing module behind compilation feature
- [x] unused variables and dead code cleanup
- [x] integration tests speedup

### CODE REVIEW

File: core/market/decentralized/src/lib.rs [hunk 1]
```diff
@@ -1,27 +1,23 @@
-// TODO: This is only temporary as long there's only market structure.
-//       Remove as soon as possible.
-#![allow(dead_code)]
-#![allow(unused_variables)]
-// #![allow(unused_imports)]
-
```
Comment (nieznanysprawiciel) [inline core/market/decentralized/src/lib.rs:6]: :+1:

File: core/market/decentralized/src/negotiation/provider.rs [hunk 2]
```diff
@@ -1,14 +1,16 @@
+// TODO: This is only temporary
+#![allow(dead_code)]
+
```
Comment (nieznanysprawiciel) [inline core/market/decentralized/src/negotiation/provider.rs:3]: Is there still any dead code in this file?

File: core/market/decentralized/tests/test_initial_proposal.rs [hunk 3]
```diff
@@ -1,418 +1,411 @@
-mod utils;
-
-#[cfg(test)]
-mod tests {
```
Comment (nieznanysprawiciel) [inline core/market/decentralized/tests/test_initial_proposal.rs:4]: I thought `mod tests` is necessary. But it isn't :D ?

File: core/market/decentralized/tests/test_initial_proposal.rs [hunk 4]
```diff
@@ -1,418 +1,411 @@
-mod utils;
```
Comment (nieznanysprawiciel) [inline core/market/decentralized/tests/test_initial_proposal.rs:1]: This is good change. Maybe compiler will stop reporting duplicated errors

File: core/market/decentralized/src/negotiation/provider.rs (hunk 2 above)
Comment (tworec) [reply core/market/decentralized/src/negotiation/provider.rs:3]: Noope. :) Removing

File: core/market/decentralized/tests/test_initial_proposal.rs (hunk 3 above)
Comment (tworec) [reply core/market/decentralized/tests/test_initial_proposal.rs:4]: yup

File: core/market/decentralized/tests/test_initial_proposal.rs (hunk 4 above)
Comment (tworec) [reply core/market/decentralized/tests/test_initial_proposal.rs:1]: it was my intention

**Pull request merged**